test/unit/providers/qrscanner_provider_test.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:superformula_test/core/service_locator.dart';
+import 'package:superformula_test/domain/qr_code/provider/scanner_provider.dart';
+
+void main() async {
+  await ServiceLocator.register();
+  group('Tests on the clear all function in the Scanner Provider', () {
+    final scannerProvider = ScannerProvider();
```

Comment:
suggestion: since we're using GetIt, we can replace reference to our `RestClient` with a mock implementation which will allow us to mock our API calls within our tests super easily.